pkg/occlient/occlient.go

```diff
@@ -892,6 +899,21 @@ func (c *Client) GetClusterServiceClasses() ([]scv1beta1.ClusterServiceClass, er
 	return classList.Items, nil
 }
 
+func (c *Client) CreateServiceInstance(componentName string, componentType string, labels map[string]string, applicationName string, projectName string) error {
+	// Patching Template with ODO specific labels
+	labelPatch := fmt.Sprintf(`{"labels": {"app.kubernetes.io/component-name": "%s", "app.kubernetes.io/component-type": "%s", "app.kubernetes.io/name": "%s"}}`, componentName, componentType, applicationName)
+	_, err := c.templateClient.Templates("openshift").Patch(componentType, types.StrategicMergePatchType, []byte(labelPatch))
+	if err != nil {
+		return errors.Wrap(err, "unable to patch template")
+	}
+	// Creating Service Instance
+	_, err = c.serviceCatalogClient.ServiceInstances(projectName).Create(&scv1beta1.ServiceInstance{metav1.TypeMeta{Kind: "ServiceInstance", APIVersion: "servicecatalog.k8s.io/v1beta1"}, metav1.ObjectMeta{Finalizers: []string{"kubernetes-incubator/service-catalog"}, Name: componentName, Namespace: "myproject", Labels: labels}, scv1beta1.ServiceInstanceSpec{UserInfo: &scv1beta1.UserInfo{Username: "developer"}, PlanReference: scv1beta1.PlanReference{ClusterServiceClassExternalName: componentType}}, scv1beta1.ServiceInstanceStatus{}})
```

Comment:
please add break `ServiceInstance` struct to multiple lines,it is hard to see what is going on there.

Reply:
You are passing labels there. If you can pass labels, why is the patching needed in the previous step?

Reply:
oh I am gonna remove this label, I was just playing around it. this label will only apply to `serviceinstance`. But we want label on every resource that will be created for example, deploymentconfig, services, routes